packages/block-editor/src/components/provider/use-block-sync.js

```diff
@@ -186,23 +177,6 @@ export default function useBlockSync( {
 		}
 	}, [ controlledBlocks, clientId ] );
 
-	const isMounted = useRef( false );
-
-	useEffect( () => {
-		// On mount, controlled blocks are already set in the effect above.
-		if ( ! isMounted.current ) {
-			isMounted.current = true;
-			return;
-		}
-
-		// When the block becomes uncontrolled, it means its inner state has been reset
-		// we need to take the blocks again from the external value property.
-		if ( ! isControlled ) {
-			pendingChanges.current.outgoing = [];
-			setControlledBlocks();
-		}
```

Comment:
So now, I have a vague memory of the Navigation block switching between "not controlled" into "controlled" to support some legacy mode of that block. @draganescu might know more. I guess the testing instructions is to try to load that old navigation block markup in the editor and ensure that everything works as expected.

Reply:
But the potentialy buggy part is when you hit Undo after this switch from uncontrolled to controlled. Then probably the navigation block should switch back to uncontrolled? And revert to the uncontrolled inner blocks specified in the markup?

That's what the title of #37484 (the PR that claims to fix #37361) says: "avoid undo issues"

Reply:
>  the navigation block should switch back to uncontrolled? 

I can't tell if that was ever the case? I don't even know if we want that, the uncontrolled version is _just a way_ for themes to include default navigation content for style reasons.